pkg/networkutils/network.go

```diff
@@ -448,22 +452,25 @@ func useExternalSNAT() bool {
 func randomizeSNAT() snatType {
 	defaultValue := randomHashSNAT
 	defaultString := "hash based random"
-	if strValue := os.Getenv(envRandomizeSNAT); strValue != "" {
-		parsedValue, err := strconv.ParseBool(strValue)
-		if err != nil {
-			if strings.Compare( "prng", strValue) == 0 {
-				return randomPRNGSNAT
-			}
-			log.Error("Failed to parse "+envRandomizeSNAT+"; using default: "+defaultString, err.Error())
-			return defaultValue
-		}
-		// true is equal to hash based random
-		if parsedValue {
-			return randomHashSNAT
-		}
-		// false is equal to sequential
+	strValue := os.Getenv(envRandomizeSNAT)
```

Comment:
Nitpick: as all values are now strings we can also use a switch over the environment variable's value. 
```go
switch os.Getenv(envRandomizeSNAT) {
case "":
	// empty means default
	return defaultValue
case "prng":
	// prng means to use --random-fully
	return randomPRNGSNAT
case "none":
	// none means to disable randomisation (no flag)
	return sequentialSNAT
case "hashrandom":
	// hashrandom means to use --random
	return randomHashSNAT
default:
	// if we get to this point, the environment variable has an invalid value
	log.Error("Failed to parse " + envRandomizeSNAT + "; using default: " + defaultString + ". Provided string was " + strValue)
	return defaultValue
}
```
It's mostly a matter of taste to prefer the succession of `if` statements or the `switch`.

Reply:
It does look neater.